src/passes/Print.cpp

```diff
@@ -835,73 +813,94 @@ struct PrintSExpression : public Visitor<PrintSExpression> {
     printName(curr->value, o) << "))";
   }
   void visitGlobal(Global* curr) {
+    doIndent(o, indent);
     o << '(';
-    printMedium(o, "global ");
-    printName(curr->name, o) << ' ';
-    if (curr->mutable_) {
-      o << "(mut " << printType(curr->type) << ") ";
+    if (curr->imported()) {
```

Comment:
How about:
```c++
doIndent(o, indent);
o << '(';
if (!curr->imported()) {
  printMedium(o, "global ");
} else {
  printMedium(o, "import ");
  printText(o, curr->module.str) << ' ';
  printText(o, curr->base.str) << ' ';
  o << "(global ";
}

printName(curr->name, o) << ' ';
if (curr->mutable_) {
  o << "(mut " << printType(curr->type) << ") ";
} else {
  o << printType(curr->type);
}

if (!curr->imported()) {
  o << ' ';
  visit(curr->init);
} else {
  o << ')'
}
o << ')';
o << maybeNewLine;
```

This adds support for printing mutable global imports (even though that's not currently allowed? We'll add it sooner or later) without duplication.

Anyway maybe that's not the best approach either, but duplicating the global-name-type printing across these branches feels weird.

Reply:
There's an open PR for mutable globals, but might as well do it here, yeah - the other PR will have a lot of conflicts anyhow, and refactoring it for clarity here is easier with support for mutable globals. (The other PR can add the tests etc.)